drivers/bridge/bridge.go

```diff
@@ -568,11 +575,113 @@ func (d *driver) DeleteEndpoint(nid, eid types.UUID) error {
 
 // Join method is invoked when a Sandbox is attached to an endpoint.
 func (d *driver) Join(nid, eid types.UUID, sboxKey string, options map[string]interface{}) error {
-	return nil
+	var err error
+	if !d.config.EnableICC {
+		err = d.link(nid, eid, options, true)
+	}
+	return err
 }
 
 // Leave method is invoked when a Sandbox detaches from an endpoint.
 func (d *driver) Leave(nid, eid types.UUID, options map[string]interface{}) error {
+	var err error
+	if !d.config.EnableICC {
+		err = d.link(nid, eid, options, false)
+	}
+	return err
+}
+
+func (d *driver) link(nid, eid types.UUID, options map[string]interface{}, enable bool) error {
+	network, err := d.getNetwork(nid)
+	if err != nil {
+		return err
+	}
+	endpoint, err := network.getEndpoint(eid)
+	if err != nil {
+		return err
+	}
+
+	if endpoint == nil {
+		return EndpointNotFoundError(eid)
+	}
+
+	cc, err := parseContainerOptions(options)
+	if err != nil {
+		return err
+	}
+	if cc == nil {
+		return nil
+	}
+
+	if endpoint.config != nil && endpoint.config.PortBindings != nil {
+		for _, p := range cc.parentEndpoints {
+			var parentEndpoint *bridgeEndpoint
+			parentEndpoint, err = network.getEndpoint(p)
+			if err != nil {
+				return err
+			}
+			if parentEndpoint == nil {
```

Comment:
Actually this is an error. When this list is provided during `Join` all of the endpoints (both parents and children) need to exist. So we should fail this case as well.